src/qibo/quantum_info/hamming_weight.py

```diff
@@ -0,0 +1,260 @@
+"Module with the object that stores results from circuit execution using the HammingWeightBackend."
+
+from typing import Optional, Union
+
+import numpy as np
+from scipy.special import binom
+
+from qibo.backends import HammingWeightBackend
+from qibo.config import raise_error
+from qibo.result import MeasurementOutcomes, QuantumState
+
+
+class HammingWeightResult(QuantumState, MeasurementOutcomes):
+    """Object storing the results of a circuit execution with the
+    :class:`qibo.backends.hamming_weight.HammingWeightBackend`.
+
+    Args:
+        state (ndarray): statevector with a fixed Hamming weight.
+            The dimension of the state is :math:`d = \\binom{n}{k}`, where
+            :math:`n` is the number of qubits and :math:`k` is the Hamming weight.
+            The components of the state are ordered in lexicographical order.
+        weight (int): Hamming weight of ``state``.
+        nqubits (int): number of qubits of ``state``.
+        measurements (list, optional): list of measurements gates :class:`qibo.gates.M`.
+            Defaults to ``None``.
+        nshots (int, optional): number of shots used for sampling the measurements.
+            Defaults to :math:`1000`.
+        engine (str, optional): engine to use in the execution of the
+            :class:`qibo.backends.HammingWeightBackend`. It accepts ``"numpy"``, ``"numba"``,
+            ``"cupy"``, and ``"cuquantum"``. If ``None``, defaults to the corresponding engine
+            from the current backend. Defaults to ``None``.
+    """
+
+    def __init__(
+        self,
+        state,
+        weight: int,
+        nqubits: int,
+        measurements=None,
+        nshots: int = 1000,
+        engine=None,
+    ):  # pylint: disable=too-many-arguments
+
+        backend = HammingWeightBackend(engine)
+        QuantumState.__init__(self, state, backend)
+        MeasurementOutcomes.__init__(self, measurements, backend)
+
+        self.measurements = measurements
+        self.nqubits = nqubits
+        self.nshots = nshots
+        self.weight = weight
+        self.n_choose_k = int(binom(self.nqubits, self.weight))
+
+        self._state = state
+        self._samples = None
+        self._probs = None
+        self._frequencies = None
+        self._measurement_gate = None
+        self._repeated_execution_frequencies = None
+        self._measurement_gate = self.measurement_gate
```

Comment:
These are also not needed as they are already assigned by the `MeasurementOutcomes.__init__(self, measurements, backend)`.